File: backend/internal/interfaces/http/hertz/dto/article.go
// Package dto 定义 HTTP 输入输出结构，不进入 Domain。
package dto

import "time"

type ArticleSource struct {
	ID      int64   `json:"id"`
	Title   string  `json:"title"`
	SiteURL *string `json:"site_url"`
}

// ArticleAuthor 只暴露稳定 ID 与昵称，不暴露登录用户名。
type ArticleAuthor struct {
	ID       string `json:"id"`
	Nickname string `json:"nickname"`
}

// RSSArticleOrigin 与 UserArticleOrigin 组成 origin 判别联合：ArticleItem.Origin 只会是其中之一。
type RSSArticleOrigin struct {
	Type              string        `json:"type"`
	Source            ArticleSource `json:"source"`
	CanonicalURL      string        `json:"canonical_url"`
	SourcePublishedAt *time.Time    `json:"source_published_at"`
}

type UserArticleOrigin struct {
	Type   string        `json:"type"`
	Author ArticleAuthor `json:"author"`
}

type ArticleItem struct {
	ID          int64     `json:"id"`
	Title       string    `json:"title"`
	Excerpt     string    `json:"excerpt"`
	PublishedAt time.Time `json:"published_at"`
	Origin      any       `json:"origin"`
}

type ArticleListResponse struct {
	Items      []ArticleItem `json:"items"`
	NextCursor *string       `json:"next_cursor"`
	HasMore    bool          `json:"has_more"`
}

// ArticleDetailResponse 在列表项基础上携带当前公开修订的清洗 HTML，字段平铺。
type ArticleDetailResponse struct {
	ArticleItem
	ContentHTML string `json:"content_html"`
}

File: web/src/types/article.ts
export interface ArticleSource {
  id: number
  title: string
  site_url: string | null
}

export interface ArticleAuthor {
  id: string
  nickname: string
}

export interface RSSArticleOrigin {
  type: 'rss'
  source: ArticleSource
  canonical_url: string
  source_published_at: string | null
}

export interface UserArticleOrigin {
  type: 'user'
  author: ArticleAuthor
}

export type ArticleOrigin = RSSArticleOrigin | UserArticleOrigin

export interface ArticleEnhancement {
  summary: string
  keywords: string[]
  topics: string[]
  generated_at: string
}

export interface ArticleItem {
  id: number
  title: string
  excerpt: string
  published_at: string
  origin: ArticleOrigin
  enhancement: ArticleEnhancement | null
}

export interface ArticlePage {
  items: ArticleItem[]
  next_cursor: string | null
  has_more: boolean
}

export type ArticleSearchPage = ArticlePage

export interface ArticleSearchQuery {
  q: string
  keyword?: string
  topic?: string
  source_id?: number
  limit?: number
  cursor?: string
}

export interface ArticleDetail extends ArticleItem {
  content_html: string
}
